spec/acceptance/keepalived_spec.rb

```diff
@@ -78,63 +40,94 @@ class { 'keepalived':
     end
 
     # Works around any timing issues
-    it 'has acquired the ip' do
-      ip_result = shell('sleep 10; ip addr')
-      expect(ip_result.stdout).to match %r{.*inet 10\.0\.0\.1/16 .*}
+    describe 'when service running' do
+      it 'has acquired the ip' do
+        ip_result = shell('sleep 10; ip addr')
+        expect(ip_result.stdout).to match %r{.*inet 10\.0\.0\.1/16 .*}
+      end
     end
   end
 
   context 'on master with globalconf' do
-    pp = <<-EOS
-    class { 'keepalived':
-      sysconf_options => '-D --vrrp',
-    }
-    class { 'keepalived::global_defs':
-      notification_email => 'nospan@example.com',
-    }
-    EOS
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-PUPPET
+        class { 'keepalived':
+          sysconf_options => '-D --vrrp',
+        }
+
+        keepalived::vrrp::instance { 'VI_50':
+          interface         => $facts['networking']['primary'],
+          state             => 'MASTER',
+          virtual_router_id => 50,
+          priority          => 101,
+          auth_type         => 'PASS',
+          auth_pass         => 'secret',
+          virtual_ipaddress => [ '10.0.0.1/16' ],
+        }
+        class { 'keepalived::global_defs':
+          notification_email => 'nospan@example.com',
+        }
+        PUPPET
+      end
+    end
 
-    it 'works with no error' do
-      apply_manifest(pp, catch_failures: true)
+    describe file('/etc/keepalived/keepalived.conf') do
+      it { is_expected.to be_file }
+      its(:content) { is_expected.to contain('notification_email').from('global_defs').to('nospan@example.com') }
     end
+  end
 
-    it 'works idempotently' do
-      apply_manifest(pp, catch_changes: true)
+  context 'with unmanaged external config' do
+    it_behaves_like 'an idempotent resource' do
+      let(:manifest) do
+        <<-PUPPET
+        file { '/etc/keepalived/myconfig.conf':
+          owner   => 'root',
+          group   => 'root',
+          mode    => '0644',
+          content => "vrrp_instance VI_50 { interface ${facts['networking']['primary']}
+          virtual_router_id 50 }",
```

Comment:
the newline in the string looks a bit odd. Is that on purpose?

Reply:
I don't know what separator keepalive use for different configuration, I've tried comma and semicolon but not work, with newline it pass the check during service start

Reply:
"...\n..." should be able to do the trick w/o wrapping the line then (I *guess*)